app/services/assign_doi.rb

```diff
@@ -0,0 +1,28 @@
+class AssignDoi
+
+  DOI_PREFIX = '10.7924'
+
+  attr_accessor :work
+
+  def self.call(work)
+    new(work).assign!
```

Comment:
As a general point, I probably in this case would simply do the action in the class method.  It does not seem that you're really gaining anything by instantiating an object.